util/serializers.py

```diff
@@ -0,0 +1,78 @@
+from rest_framework import serializers
+from django.contrib.auth import get_user_model
+from projects.models import Project
+from allocations.models import Allocation
+
+# from allocations.allocations_api import BalanceServiceClient
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = get_user_model()
+
+        fields = [
+            "username",
+            "first_name",
+            "last_name",
+            "email",
+        ]
+
+
+class AllocationSerializer(serializers.ModelSerializer):
+    """Serialize Allocation object"""
+
+    """
+    keys in the array "fields" can be queried in URL by "?key=value"
+    Their format can be overridden by defining it outside the "Meta" class.
+    """
+
+    class Meta:
+        model = Allocation
+        fields = [
+            "id",
+            "project",
+            "status",
+            "start_date",
+            "expiration_date",
+            "su_used",
+            "requestor",
+            "date_requested",
+            "su_requested",
+            "justification",
+            "reviewer",
+            "date_reviewed",
+            "su_allocated",
+            "decision_summary",
+        ]
+
+
+class NestedAllocationSerializer(AllocationSerializer):
+    requestor = UserSerializer()
+    reviewer = UserSerializer()
+
+
+class ProjectSerializer(serializers.ModelSerializer):
+    """Serialize project object"""
+
+    class Meta:
+        model = Project
+        fields = [
+            "charge_code",
+            "title",
+            "nickname",
+            "type",
+            "project_type",
+            "field",
+            "description",
+            "pi",
+            "allocations",
+        ]
```

Comment:
The serializers are the big part of the motivation here. It's most of the hard part of what we've been doing in the "allocation mapper" functions, but formalized, and can automatically infer these properties from the defined models.

This lets us keep all the translation and validation logic in one place.

By default, it will use the primary key to reference foreign keys.

If we define create() or update() methods, these will enable validated write access back to the django DB.